pkg/cmd/ephemeralenvironment/deprovision-environment/deprovision_environment.go

```diff
@@ -0,0 +1,181 @@
+package deprovision_environment
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/MakeNowJust/heredoc/v2"
+	"github.com/OctopusDeploy/cli/pkg/cmd"
+	"github.com/OctopusDeploy/cli/pkg/cmd/ephemeralenvironment/util"
+	"github.com/OctopusDeploy/cli/pkg/constants"
+	"github.com/OctopusDeploy/cli/pkg/factory"
+	"github.com/OctopusDeploy/cli/pkg/output"
+	"github.com/OctopusDeploy/cli/pkg/question/selectors"
+	"github.com/OctopusDeploy/cli/pkg/util/flag"
+	"github.com/OctopusDeploy/go-octopusdeploy/v2/pkg/environments/v2/ephemeralenvironments"
+	"github.com/spf13/cobra"
+)
+
+const (
+	FlagName = "name"
+)
+
+type DeprovisionEnvironmentFlags struct {
+	Name *flag.Flag[string]
+}
+
+func NewDeprovisionEnvironmentFlags() *DeprovisionEnvironmentFlags {
+	return &DeprovisionEnvironmentFlags{
+		Name: flag.New[string](FlagName, false),
+	}
+}
+
+type DeprovisionEnvironmentOptions struct {
+	*DeprovisionEnvironmentFlags
+	*cmd.Dependencies
+	Command                     *cobra.Command
+	GetAllEphemeralEnvironments func() ([]*ephemeralenvironments.EphemeralEnvironment, error)
+}
+
+func NewDeprovisionEnvironmentOptions(deprovisionEnvironmentFlags *DeprovisionEnvironmentFlags, dependencies *cmd.Dependencies, command *cobra.Command) *DeprovisionEnvironmentOptions {
+	return &DeprovisionEnvironmentOptions{
+		DeprovisionEnvironmentFlags: deprovisionEnvironmentFlags,
+		Dependencies:                dependencies,
+		Command:                     command,
+		GetAllEphemeralEnvironments: func() ([]*ephemeralenvironments.EphemeralEnvironment, error) {
+			return getAllEphemeralEnvironments(dependencies)
+		},
+	}
+}
+
+func getAllEphemeralEnvironments(dependencies *cmd.Dependencies) ([]*ephemeralenvironments.EphemeralEnvironment, error) {
+	allEphemeralEnvironments, err := ephemeralenvironments.GetAll(dependencies.Client, dependencies.Client.GetSpaceID())
+
+	if err != nil {
+		return nil, err
+	}
+
+	return allEphemeralEnvironments.Items, nil
+}
+
+func NewCmdDeprovisionEnvironment(factory factory.Factory) *cobra.Command {
+	deprovisionEnvironmentFlags := NewDeprovisionEnvironmentFlags()
+
+	command := &cobra.Command{
+		Use:     "deprovision-environment",
+		Short:   "Deprovision an environment",
+		Long:    "Deprovision an environment",
+		Example: heredoc.Docf("$ %s ephemeral-environment deprovision-environment --name PR-1234", constants.ExecutableName),
+		RunE: func(c *cobra.Command, _ []string) error {
+			deprovisionEnvironmentOptions := NewDeprovisionEnvironmentOptions(deprovisionEnvironmentFlags, cmd.NewDependencies(factory, c), c)
+
+			return DeprovisionEnvironmentRun(deprovisionEnvironmentOptions, c)
+		},
+	}
+
+	flags := command.Flags()
+	flags.StringVarP(&deprovisionEnvironmentFlags.Name.Value, FlagName, "n", "", "Name of the environment")
+
+	return command
+}
+
+func DeprovisionEnvironmentRun(deprovisionEnvironmentOptions *DeprovisionEnvironmentOptions, command *cobra.Command) error {
```

Comment:
Also I know I called out the names being short, but looking at other code, I think this could be opts to be consistent with other commands